code/game/machinery/mind_machine.dm

```diff
@@ -0,0 +1,856 @@
+		errorMessage = SHARD_REACTION
+	currentlySwapping = TRUE
+	icon_state = "mind_hub_active"
+	connectOne.icon_state = "mind_pod_active"
+	connectTwo.icon_state = "mind_pod_active"
+	if(is_type_in_list(occupantOne, illegalSwap) || is_type_in_list(occupantTwo, illegalSwap))
+		malfSwap = TRUE //fun sanity check. Can't send their mind into the illegal body.
+	flyTally(occupantOne)
+	flyTally(occupantTwo)
+	for(var/prog in 1 to 40/manipRating) //Counts up 5 to 10 seconds, checks if we lost power each time and ruins your day if so. Otherwise just for UI progress bar
+		if(stat & NOPOWER)
+			malfSwap = TRUE
+		if(!connectOne.Adjacent(src) || !connectTwo.Adjacent(src))
+			malfSwap = TRUE
+		sleep(1 SECONDS)
+		if(gcDestroyed)
+			return
+		if(!locate(occupantOne,connectOne.contents) || !locate(occupantTwo, connectTwo.contents) || !connectOne || !connectTwo)
+			errorMessage = MISSING_OCC
+			swapProgress = 0
+			currentlySwapping = FALSE
+			occupantOne = null
+			occupantTwo = null
+			return
+		swapProgress += 1
+		nanomanager.update_uis(src)
+	if(malfSwap) ///////////Swaps begin///////
+		malfunction()
+		return
+	if(mindTypeOne == LOWER && mindTypeTwo == LOWER)
+		lowerSwap()
+		return
+	if(theFly.len > 0)
+		theFlySwap()
+		return
+	if(mindTypeOne == HIGHER && mindTypeTwo == HIGHER)
+		higherSwap()
+		return
+	if(mindTypeOne ==HIGHER && mindTypeTwo == LOWER || mindTypeOne == LOWER && mindTypeTwo == HIGHER)
+		animorphsSwap()
+		return
+	if(mindTypeOne == SILICON && scanRatingOne <7 || mindTypeTwo == SILICON && scanRatingTwo <7)
+		errorMessage = SCANNER_INSUFF
+		swapProgress = 0
+		currentlySwapping = FALSE
+		connectOne.icon_state = "mind_pod_closed"
+		connectTwo.icon_state = "mind_pod_closed"
+		icon_state = "mind_hub"
+		return
+	if(mindTypeOne == SILICON && mindTypeTwo == SILICON)
+		higherSwap() //Silicon swaps, checks if their pod is upgraded enough then performs normal swap
+		return
+	if(mindTypeOne == SILICON && mindTypeTwo != SILICON)
+		if(mindTypeTwo == LOWER)
+			animorphsSwap()
+			return
+		if(mindTypeTwo == HIGHER && theFly.len == 0)
+			higherSwap()
+			return
+	if(mindTypeTwo == SILICON && mindTypeOne != SILICON)
+		if(mindTypeOne == LOWER)
+			animorphsSwap()
+			return
+		if(mindTypeOne == HIGHER && !theFly.len)
+			higherSwap()
+			return
+	else
+		errorMessage = UNKNOWN_INT
+		currentlySwapping = FALSE
+		swapProgress = 0
+		unlockPods()
+
+/obj/machinery/mind_machine/mind_machine_hub/proc/flyTally(var/mob/living/M)
+	for(var/obj/item/weapon/holder/i in get_contents_in_object(M))  //Carp in pack? Outta luck, Jack
+		theFly += i
+	var/list/borers = M.get_brain_worms()
+	for(var/mob/living/simple_animal/borer/B in borers) //borers in the body are counted
+		theFly += B
+
+/obj/machinery/mind_machine/mind_machine_hub/emp_act(severity)
+	..()
+	if(currentlySwapping)
+		malfSwap = TRUE
+		return
+
+/obj/machinery/mind_machine/mind_machine_hub/emag_act()
+	if(currentlySwapping)
+		spark(src)
+		malfSwap = TRUE
+		return
```

Comment:
why do you keep returning

Reply:
Because 90% of this was written when I barely knew what return did. I now feel a tinge of shame.